heracal/tests/test_omni.py

```diff
@@ -84,6 +84,42 @@ def test_compute_reds(self):
                 for l in k:
                     nt.assert_true(isinstance(l, omni.Antpol))
 
+    def test_minVreds(self):
+        reds = omni.compute_reds(4, self.pols, self.info.antloc[:self.info.nant])
+        mVreds = omni.minVreds(reds)
+        # test that the new reds array is shorter by 1/4, as expected
+        nt.assert_equal(len(mVreds),len(reds) - len(reds)/4)
+        # test that we haven't invented or lost baselines
+        cr,cmv = 0,0
+        for arr in reds: cr+=len(arr)
+        for arr in mVreds: cmv+=len(arr)
+        nt.assert_equal(cr,cmv)
+        # test that no crosspols are in linpol red arrays and vice versa
+        for arr in mVreds:
+            p0 = arr[0][0].pol()+arr[0][1].pol()
+            for ap in arr:
+                p = ap[0].pol()+ap[1].pol()
+                nt.assert_equal(len(set(p)),len(set(p0)))
+        # test that every xy has it's corresponding yx in the array
+        for arr in mVreds:
+            p0 = arr[0][0].pol()+arr[0][1].pol()
+            if len(set(p0))==1: continue #not interested in linpols
+            for ap in arr:
+                ai,aj = ap
+                bi,bj = omni.Antpol(ai.ant(),aj.pol(),self.info.nant),omni.Antpol(aj.ant(),ai.pol(),self.info.nant) 
+                if (bi,bj) in arr:
+                    continue
+                elif (bj,bi) in arr:
+                    continue
+                else:
+                    raise Exception('Something has gone wrong in polarized redundancy calculation (missing crosspols)')
+        # test AssertionError
+        _reds = reds[:-1]
+        try:
+            omni.minVreds(_reds)
+        except:
+            pass   
```

Comment:
Never never never `except: pass`!!!

For one thing, this test can never fail. If no exception is raised nothing will happen.

There is probably a nice way to express "this operation should cause an exception" in Python tests, but at least:

```
try:
   omni.minVreds(_reds)
   assert False, 'should not have gotten here'
except Exception:
   pass
```

Also it would probably be better for `minVreds` to raise a `ValueError` rather than a generic exception since that's the kind of exception you're supposed to use when a function has been given invalid inputs.